c/cmake/Toolchain-gcc-arm-embedded.cmake

```diff
@@ -41,21 +41,26 @@ message(STATUS "Cross-compiling with the gcc-arm-embedded toolchain")
 message(STATUS "Toolchain prefix: ${CMAKE_INSTALL_PREFIX}")
 
 set(CMAKE_FIND_ROOT_PATH  ${CMAKE_INSTALL_PREFIX})
-
 set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
 set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
-
 set(CMAKE_C_FLAGS "-fno-common -ffunction-sections -fdata-sections")
-
 if (CMAKE_SYSTEM_PROCESSOR STREQUAL "cortex-m4")
 
   set(CMAKE_C_FLAGS
     "${CMAKE_C_FLAGS}"
     "-mcpu=cortex-m4 -march=armv7e-m -mthumb"
     "-mfloat-abi=hard -mfpu=fpv4-sp-d16"
   )
+  
+elseif (CMAKE_SYSTEM_PROCESSOR STREQUAL "cortex-m7")
 
+  set(CMAKE_C_FLAGS
+    "${CMAKE_C_FLAGS}"
+    "-mcpu=cortex-m7 -march=armv7e-m -mthumb"
```

Comment:
`-march=armv7e-m` isn't a flag in the CubeMX `Makefile`.. should be enforced by `mcpu` https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html